front-end-tools/src/Main.tsx

```diff
@@ -1081,19 +1445,974 @@ export default function Main(props: ConnectionProps) {
+                                                        {getObjectExample(entryPointTemplateReadFunction)}
+                                                    </textarea>
+                                                )}
+
+                                                {attributeForm.formState.errors.inputParameter && (
+                                                    <Alert key="info" variant="info">
+                                                        {' '}
+                                                        Input parameter is required{' '}
+                                                    </Alert>
+                                                )}
+                                                <Form.Text />
+                                            </Form.Group>
+                                        )}
+
+                                        {parsingErrorRead !== undefined && (
+                                            <Alert variant="danger"> Error: {parsingErrorRead}. </Alert>
+                                        )}
+                                    </div>
+                                )}
+
+                                <br />
+
+                                <Button
+                                    variant="primary"
+                                    type="button"
+                                    onClick={attributeForm.handleSubmit((data) => {
+                                        setReadError(undefined);
+                                        setReturnValue(undefined);
+                                        const promise = read(
+                                            client,
+                                            data.smartContractName,
+                                            BigInt(data.smartContractIndex),
+                                            data.entryPointName,
+                                            data.deriveFromSmartContractIndex
+                                                ? embeddedModuleSchemaBase64Read
+                                                : uploadedModuleSchemaBase64Read,
+                                            data.inputParameter,
+                                            data.inputParameterType,
+                                            data.hasInputParameter,
+                                            data.deriveFromSmartContractIndex
+                                        );
+
+                                        promise
+                                            .then((value) => {
+                                                setReturnValue(value);
+                                            })
+                                            .catch((err: Error) => setReadError((err as Error).message));
+                                    })}
+                                >
+                                    Read Smart Contract
+                                </Button>
```

Comment:
Idiomatic way to do this would be to:

```tsx
const {handleSubmit} = useForm(...);
function onSubmit(values: FormValues) { // or whatever the type of the form is?
...
}
...
<form onSubmit={handleSubmit(onSubmit)}>
    ... // form fields
    <Button type="submit" ...>...</Button>
</form>
```